traefik_mesh/tests/test_unit.py

```diff
@@ -0,0 +1,108 @@
+# (C) Datadog, Inc. 2024-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import mock
+import pytest
+
+from datadog_checks.base.constants import ServiceCheck
+from datadog_checks.dev.utils import assert_service_checks, get_metadata_metrics
+from datadog_checks.traefik_mesh import TraefikMeshCheck
+
+from .common import (
+    OM_METRICS,
+    OM_MOCKED_INSTANCE,
+    OM_MOCKED_INSTANCE_CONTROLLER,
+    OPTIONAL_METRICS,
+    get_fixture_path,
+    read_json_fixture,
+)
+
+
+def test_check_mock_traefik_mesh_openmetrics(dd_run_check, aggregator, mock_http_response):
+    mock_http_response(file_path=get_fixture_path('traefik_proxy.txt'))
+    check = TraefikMeshCheck('traefik_mesh', {}, [OM_MOCKED_INSTANCE])
+    dd_run_check(check)
+
+    for metric in OM_METRICS:
+        if metric not in OPTIONAL_METRICS:
+            aggregator.assert_metric(metric)
+            aggregator.assert_metric_has_tag(metric, 'test:traefik_mesh')
+
+    aggregator.assert_all_metrics_covered()
+    aggregator.assert_metrics_using_metadata(get_metadata_metrics())
+    aggregator.assert_service_check('traefik_mesh.openmetrics.health', ServiceCheck.OK)
+    assert_service_checks(aggregator)
+
+
+def test_check_mock_invalid_traefik_mesh_openmetrics(dd_run_check, aggregator, mock_http_response):
+    mock_http_response(status_code=503)
+    check = TraefikMeshCheck('traefik_mesh', {}, [OM_MOCKED_INSTANCE])
+    with pytest.raises(Exception):
```

Comment:
Could we assert something about the exception here? I know we can't use the exception type, but let's use the `match=` argument ([docs](https://docs.pytest.org/en/4.6.x/reference.html#pytest.raises)) to at least filter out the exception message.